test/TesterInternal/ActivationsLifeCycleTests/DeactivateOnIdleTests.cs

```diff
@@ -353,23 +361,35 @@ public void Dispose()
             }
 
             // Now we know that there's an activation; try both silos and deactivate it incorrectly
-            int primaryActivation = await testCluster.Primary.TestHook.UnregisterGrainForTesting(grainReference);
+            int primaryActivation =
+                await
+                    testCluster.Primary.GetTestHook(this.testCluster.InternalGrainFactory)
+                        .UnregisterGrainForTesting(grainReference);
             int secondaryActivation = 0;
 
             if (isMultipleSilosPresent)
             {
-                secondaryActivation = await testCluster.SecondarySilos[0].TestHook.UnregisterGrainForTesting(grainReference);
+                secondaryActivation =
+                    await
+                        testCluster.SecondarySilos[0].GetTestHook(this.testCluster.InternalGrainFactory)
```

Comment:
API-wise, it could be nicer if this was testCluster.GetTestHook(SecondarySilos[0]).XXX, since the internal grain factory is an implementation detail, and also there are some test hooks that can be useful for end users who do not have InternalsVisibleTo in order to get ahold of this internal class.